crates/chisel/bin/main.rs

```diff
@@ -228,3 +228,48 @@ impl Provider for ChiselParser {
         Ok(Map::from([(Config::selected_profile(), Dict::default())]))
     }
 }
+
+/// Evaluate a single Solidity line.
+async fn dispatch_repl_line(dispatcher: &mut ChiselDispatcher, line: &str) {
+    match dispatcher.dispatch(line).await {
+        DispatchResult::Success(msg) | DispatchResult::CommandSuccess(msg) => if let Some(msg) = msg {
+            println!("{}", Paint::green(msg));
+        },
+        DispatchResult::UnrecognizedCommand(e) => eprintln!("{e}"),
+        DispatchResult::SolangParserFailed(e) => {
+            eprintln!("{}", Paint::red("Compilation error"));
+            eprintln!("{}", Paint::red(format!("{e:?}")));
+        }
+        DispatchResult::FileIoError(e) => eprintln!("{}", Paint::red(format!("⚒️ Chisel File IO Error - {e}"))),
+        DispatchResult::CommandFailed(msg) | DispatchResult::Failure(Some(msg)) => eprintln!("{}", Paint::red(msg)),
+        DispatchResult::Failure(None) => eprintln!("{}\nPlease Report this bug as a github issue if it persists: https://github.com/foundry-rs/foundry/issues/new/choose", Paint::red("⚒️ Unknown Chisel Error ⚒️")),
+    }
+}
+
+/// Evaluate multiple Solidity source files contained within a
+/// Chisel prelude directory.
+async fn evaluate_prelude(
+    dispatcher: &mut ChiselDispatcher,
+    maybe_prelude: Option<PathBuf>,
+) -> eyre::Result<()> {
+    if let Some(prelude_dir) = maybe_prelude {
+        let prelude_sources = fs::files_with_ext(prelude_dir, "sol");
+        let print_success_msg = !prelude_sources.is_empty();
+
+        for source_file in prelude_sources {
+            println!(
+                "{} {}",
+                Paint::yellow("⏳ Loading prelude source file:"),
```

Comment:
I'd say let's avoid emojis on the output—might not display accurately on all platforms!